boa/src/builtins/number/mod.rs

```diff
@@ -47,19 +44,22 @@ impl Number {
     /// Helper function that converts a Value to a Number.
     #[allow(clippy::wrong_self_convention)]
     fn to_number(value: &Value) -> Value {
-        match *value.deref().borrow() {
+        match value.data() {
             ValueData::Boolean(b) => {
-                if b {
+                if *b {
                     Value::from(1)
                 } else {
                     Value::from(0)
                 }
             }
             ValueData::Symbol(_) | ValueData::Undefined => Value::from(f64::NAN),
-            ValueData::Integer(i) => Value::from(f64::from(i)),
-            ValueData::Object(ref o) => (o).deref().borrow().get_internal_slot("NumberData"),
+            ValueData::Integer(i) => Value::from(f64::from(*i)),
+            ValueData::Object(ref o) => match (o).deref().borrow().data {
+                ObjectData::Number(num) => Value::from(num),
+                _ => unreachable!(),
```

Comment:
I would add a small text in the `unreachable!("")` macro for the panic, in case it ever happens.

Reply:
That was not spec compliant, made it more spec compliant here [https://github.com/boa-dev/boa/pull/419/commits/49e104b755f321bac6c4d636f74497190a827318 ](https://github.com/boa-dev/boa/pull/419/commits/49e104b755f321bac6c4d636f74497190a827318)